test/unit/lexer/location_offset.d

```diff
@@ -0,0 +1,482 @@
+module lexer.location_offset;
+
+import dmd.lexer : Lexer;
+import dmd.tokens : TOK;
+
+import support : afterEach;
+import lexer.diagnostic_reporter : deinitializeFrontend;
+
+@("first token in the source code")
+unittest
+{
+    enum code = "token";
+
+    scope lexer = new Lexer("test.d", code.ptr, 0, code.length, 0, 0);
+
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 0, code);
+}
+
+@("first token when begoffset is not 0")
+unittest
+{
+    enum code = "ignored_token token";
+
+    scope lexer = new Lexer("test.d", code.ptr, 13, code.length - 14, 0, 0);
+
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 14, code);
+}
+
+@("last token in the source code")
+unittest
+{
+    enum code = "token1 token2 token3";
+
+    scope lexer = new Lexer("test.d", code.ptr, 0, code.length, 0, 0);
+
+    lexer.nextToken;
+    lexer.nextToken;
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 14, code);
+}
+
+@("last token in the source code")
+unittest
+{
+    enum code = "token";
+
+    scope lexer = new Lexer("test.d", code.ptr, 0, code.length, 0, 0);
+
+    lexer.nextToken;
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 5, code);
+}
+
+@("identifier after block comment")
+unittest
+{
+    enum code = "/* comment */ token";
+
+    scope lexer = new Lexer("test.d", code.ptr, 0, code.length, 0, 0);
+
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 14, code);
+}
+
+@("identifier after line comment")
+unittest
+{
+    enum code = "// comment\ntoken";
+
+    scope lexer = new Lexer("test.d", code.ptr, 0, code.length, 0, 0);
+
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 11, code);
+}
+
+@("identifier after nesting block comment")
+unittest
+{
+    enum code = "/+ comment +/ token";
+
+    scope lexer = new Lexer("test.d", code.ptr, 0, code.length, 0, 0);
+
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 14, code);
+}
+
+@("token after Unix line ending")
+unittest
+{
+    enum code = "line\ntoken";
+
+    scope lexer = new Lexer("test.d", code.ptr, 0, code.length, 0, 0);
+
+    lexer.nextToken;
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 5, code);
+}
+
+@("token after Windows line ending")
+unittest
+{
+    enum code = "line\r\ntoken";
+
+    scope lexer = new Lexer("test.d", code.ptr, 0, code.length, 0, 0);
+
+    lexer.nextToken;
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 6, code);
+}
+
+@("token after Mac line ending")
+unittest
+{
+    enum code = "line\rtoken";
+
+    scope lexer = new Lexer("test.d", code.ptr, 0, code.length, 0, 0);
+
+    lexer.nextToken;
+    lexer.nextToken;
+
+    assert(lexer.token.loc.offset == 5, code);
+}
+
+@("multibyte character token")
+unittest
+{
+    enum code = "🍺";
```

Comment:
This is not a character token. It's probably an identifier token. You need single quotes (`''`) to make it a character literal.